### DISCUSSION TITLE
Guidance on making RFQuack more flexible

### DISCUSSION THREAD

**Discussion by ResistanceIsUseless** [rfquack/RFQuack#17] [General]
After dealing with a bunch of issues with setup I'm wondering if there are some enhancements / changes that can be made to make setup easier.

1. Use Github actions to create firmware releases. It looks like you we're already working on an action to run `make proto`. I'm suggesting to do that as well as build some of the more common configurations like CC1101 + ESP32, EvilCrowRFv2, etc. 
2. Provide guidance on using RFQuack as a library in Arduino. Adding a zip file that can be included in releases would be nice. 
3. Update documentation. Right now there are a bunch of gaps that require some guessing and digging through the source code to figure out.

I know you have been clear about your availability to manage this project. I'm about 90% finished with 1. but being able to get your guidance occasionally would be really helpful. I don't mind doing the work and the learning that come with it but sometimes getting some questions answered would really speed things up, preferably over discord. It would be a shame to see this project die, I feel like it has so much potential.

**Comment by phretor** [comment]
@ResistanceIsUseless thanks. This is what I have on my mental roadmap, in decreasing order of priority. Probably I should make milestones and/or properly labeled issues to keep track of these:

- create a documentation site using docbook or similar generators, even better if the doc is integrated in the source code so we don't need to document in multiple places or care for consistency.
- create self-contained examples that can be compiled in a CI/CD environment: not sure if it's better to do this with the Arduino vs. PIO CLI, and I accept recommendations.
- auto-releasing a firmware built for each combination of parameters (e.g., radio, MCU, port, transport) via Actions would then be a simple extension of the previous point
- port new drivers by creating thin wrappers from RadioLib
- add needed hardware daughterboards or adapters as needed by the previous point
- make a 3D case for the tripler version
- include a crosspoint switch in the main board to set I/O and IRQ routing in software thus eliminate the need for soldering

**Reply by ResistanceIsUseless** [reply]
I love it all! Sounds like great ideas and I'm excited to help. I can help with creating documentation and using github pages to host it. I also can help with creating firmware releases actions templates and 3D printing (I have an older rostock max but it still works!). 

I'm not experienced with circuit design or embedded development though. They are on my want to learn bucket list though. 😆 So I definitely want to learn more in those areas. I spend my days as a DevSecOps / Security Architect and this is a nice change from that.

Anyways, expect a pull request in the near future from me. I have some enhancement ideas as well but I'll wait till I get to know RFQuack better before proposing them.
